### ISSUE TITLE
compilation fails on i686 ( 32 bits )

### ISSUE THREAD

**Issue by mcirsta** [bcgsc/abyss#68]
When compiling abyss 1.5.2 on i686 I get the following error:

```
In file included from SequenceCollection.cpp:7:0:
../Common/StringUtil.h: In function 'size_t SIToBytes(std::istringstream&)':
../Common/StringUtil.h:142:23: error: left shift count >= width of type [-Werror]
    size *= (size_t)1<<40; break;
```

This is probably because size_t is 32 bits only so left shifting with 40 bits will not have the desired effects.
To fix this I suggest using uint64_t instead.

**Comment by mcirsta** [comment]
After fixing some type problems ( see pull request ) I'm getting this error:

g++ -Wall -Wextra -Werror -Wno-unused-local-typedefs  -Wl,--hash-style=both -o abyss-fac abyss_fac-fac.o libdatalayer.a ../Common/libcommon.a -ldl -lm 
../Common/libcommon.a(libcommon_a-Sequence.o): In function `reverseComplement(std::string const&)':
Sequence.cpp:(.text+0x16f): undefined reference to `opt::colourSpace'
../Common/libcommon.a(libcommon_a-Sequence.o): In function `codeToBase(unsigned char)':
Sequence.cpp:(.text+0x29a): undefined reference to `opt::colourSpace'

**Comment by benvvalk** [comment]
Thanks @mcirsta for the bug report and the associated patch.  I will look into it.

**Comment by benvvalk** [comment]
Fixed in 1.9.0 release.
